docs/faqs.md

```diff
@@ -0,0 +1,153 @@
+For example, consider Shor's algorithm: when coded up in Python, we will instinctively use
+structure such as if statements, for loops, etc., rather than writing out the finalized list of
+gates to be applied. This structure results in a *compressed* program representation compared to
+the unrolled list of quantum instructions, which is both more efficient to compile, and may even
+allow for optimizations which may not be possible when the structure is lost/the program is flattened.
+
+So ideally, we should design compiler passes to act on
+such compressed representations! In extreme cases, if the program representation does not increase in size as the program scales,
+[compilation may be constant-time](https://arxiv.org/abs/2504.12449).
+
+There is still a lot of work to do when it comes to compiling structured programs:
+
+- Often, porting known state-of-the-art compilation techniques to apply to structured programs can
+  be highly non-trivial.
+
+- There is tension between maximal optimization and compact program representation. There are still
+  a lot of questions we need to consider and resolve when it comes to compiling dynamic, structured
+  programs. For example, the more compact the program representation (which is great for
+  compilation scalability!) the less information we potentially have up-front to maximally optimize
+  programs at compile time. How do we deal with such scenarios?
+
+## How did you decide what goes into the schema?
+
+## Why Cap'n Proto?
+
+We use [Cap'n Proto](https://capnproto.org/) as our data serialization format for Jeff.
+
+**Reasons for choosing Cap'n Proto:**
+- Fast, efficient binary encoding
+- Zero-copy deserialization for high performance
+- Simple and expressive schema language
+- Robust tooling and support for key quantum programming languages (Python, C++, Rust)
+
+**Alternative serialization formats:**  
+Other models considered include [Protocol Buffers](https://protobuf.dev/), [Flatbuffers](https://flatbuffers.dev/), [MessagePack](https://msgpack.org/), and JSON. While these formats offer similar features, Cap'n Proto provides the best combination of speed, schema support, and language integration for our needs.
+
+
+## Isn't this just *another* format standard?
+
+The danger of introducing new exchange formats or standards is that they themselves may slow down
+development — perhaps because of a governance structure with too much process and/or stakeholders,
+or simply because the format is too rigid and doesn't permit extensibility for new ideas.
+
+This is something we wish to avoid — Jeff is *not* a standard we need all stakeholders to adopt, but
+an exchange format localized to compilers, designed to
+be easily extensible. Additional information for program representation and compilation can be added
+directly, and will be taken into account by software frameworks that understand the direction.
+Further, development will happen out in the open, on GitHub.
```

Comment:
I'm not convinced that open development alone will avoid the problems mentioned here. Is there a broader strategy to get _feedback_ from stakeholders and/or competing IRs, without binding them to use the format or requesting them to steer development?

As a side note... I've facilitated discussions at compilation workshops to try and determine what we need / don't need as part of a unified representation. These discussions never converge :sweat_smile: There is always something missing that someone wants. An important question to consider is, does there exist a _truly minimal_ set of requirements for an exchange format such that any (known) quantum algorithm can be compiled to Jeff from framework A, and faithfully loaded into framework B?